cmd/goauth/main.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/artifact-registry-go-tools/pkg/auth"
+)
+
+const help = `
+Handle Go authentication with Google Cloud Artifact Registry Go Repositories.
+
+Add to your GOAUTH environment variable:
+
+  export GOAUTH="sh -c 'GOPROXY=direct go run github.com/GoogleCloudPlatform/artifact-registry-go-tools/cmd/goauth@latest <location>'"
+
+To support multiple locations, add the command multiple times to the GOAUTH variable (semicolon-separated).
+
+For more details, see https://pkg.go.dev/cmd/go@master#hdr-GOAUTH_environment_variable`
+
+func main() {
+	jsonKey := flag.String("json_key", "", "path to the json key of the service account used for this location. Leave empty to use the oauth token instead.")
```

Comment:
I was wondering if we could reuse the same program entry point. The main motivation is to avoid duplicate code (the two flags here), or at least make it harder for future developers miss one another when adding more features to goauth or netrc.

maybe

```
export GOAUTH="GOPROXY=direct go run github.com/GoogleCloudPlatform/artifact-registry-go-tools/cmd/auth@latest goauth ---location"
```

Reply:
I think it's better to have it separate, because the application behavior is dictated by Go and will have to be different from the `auth` command behavior: Things like writing and reading from stdin/out and getting an additional commandline argument.

Even if they seem to be similar now, they probably will divert more in the future.